sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets

```diff
@@ -225,122 +253,151 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
     -->
 
-  <PropertyGroup>
-    <CorePublishDependsOn>
-      _InitializeDeployOnBuildProperties;
-      Publish;
-      $(_DotNetPublishFiles);
-    </CorePublishDependsOn>
-  </PropertyGroup>
-
-  <Target Name="_GenerateFunctionsAndCopyContentFiles">
-
     <PropertyGroup>
-      <OutputFile>$(PublishDir)\worker.config.json</OutputFile>
-      <ExtensionsCsProjFilePath>$([System.IO.Path]::Combine($([System.IO.Path]::GetTempPath()), $([System.IO.Path]::GetRandomFileName())))</ExtensionsCsProjFilePath>
+        <CorePublishDependsOn>
+            _InitializeDeployOnBuildProperties;
+            Publish;
+            $(_DotNetPublishFiles);
+        </CorePublishDependsOn>
     </PropertyGroup>
 
-    <GenerateFunctionMetadata
-      AssemblyPath="$(PublishDir)\$(AssemblyName).dll"
-      ReferencePaths="@(ReferencePath)"
-      ExtensionsCsProjFilePath="$(ExtensionsCsProjFilePath)"
-      AzureFunctionsVersion="$(AzureFunctionsVersion)"
-      OutputPath="$(PublishDir)"/>
-
-    <Copy
-      SourceFiles="$(_FunctionsMetadataLoaderExtensionFile)"
-      DestinationFolder="$(ExtensionsCsProjFilePath)\publishout"
-      OverwriteReadOnlyFiles="true" />
-
-    <WriteLinesToFile
-        Condition="$(SelfContained)"
-        File="$(OutputFile)"
-        Lines="$([System.IO.File]::ReadAllText($(_FunctionsWorkerConfigInputFile))
+    <Target Name="_GenerateFunctionsAndCopyContentFiles" DependsOnTargets="_GenerateFunctionsAndCopyContentFilesNetApp;_GenerateFunctionsAndCopyContentFilesNetFx" />
+
+    <Target Name="_GenerateFunctionsAndCopyContentFilesNetFx" Condition="$(TargetFrameworkIdentifier) == '.NETFramework'">
+
+        <PropertyGroup>
+            <OutputFile>$(PublishDir)\worker.config.json</OutputFile>
+            <ExtensionsCsProjFilePath>$([System.IO.Path]::Combine($([System.IO.Path]::GetTempPath()), $([System.IO.Path]::GetRandomFileName())))</ExtensionsCsProjFilePath>
+        </PropertyGroup>
+
+        <GenerateFunctionMetadata
+          AssemblyPath="$(PublishDir)\$(AssemblyName).exe"
+          ReferencePaths="@(ReferencePath)"
+          ExtensionsCsProjFilePath="$(ExtensionsCsProjFilePath)"
+          AzureFunctionsVersion="$(AzureFunctionsVersion)"
+          OutputPath="$(PublishDir)"/>
+
+        <Copy
+          SourceFiles="$(_FunctionsMetadataLoaderExtensionFile)"
+          DestinationFolder="$(ExtensionsCsProjFilePath)\publishout"
+          OverwriteReadOnlyFiles="true" />
+
+        <WriteLinesToFile
+            File="$(OutputFile)"
+            Lines="$([System.IO.File]::ReadAllText($(_FunctionsWorkerConfigInputFile))
+                                  .Replace('$functionExe$', '{WorkerRoot}$(TargetName).exe')
+                                  .Replace('$functionWorker$', '$(TargetName).exe'))"
+            Overwrite="true" />
+    </Target>
+
+    <Target Name="_GenerateFunctionsAndCopyContentFilesNetApp" Condition="$(TargetFrameworkIdentifier) == '.NETFramework'">
```

Comment:
Question: Should the file extensions used in the `WriteLinesToFile` below (312,320) be fixed from .dll to .exe?

Reply:
No, but your comment made me realize that the condition here was incorrectly checking for equality.